firmware/Core/Src/telecommands/uart_telelecommand_defs.c

```
@@ -0,0 +1,313 @@
+                    "Transmitted %u byte(s) to %s.", 
+                    tx_buffer_len, 
+                    arg_uart_port_name
+                );
+                LOG_message(
+                    LOG_source, LOG_SEVERITY_NORMAL, LOG_SINK_ALL,
+                    "Received %u byte(s) from %s: %s",
+                    rx_buffer_len,
+                    arg_uart_port_name,
+                    rx_buffer_str
+                );
+                return 0;   // Success
+            case 2:
+                LOG_message(
+                    LOG_source, LOG_SEVERITY_ERROR, LOG_SINK_ALL,
+                    "Failed UART transmission."
+                );
+                return 3;   // Error code: Error transmitting data
+            case 3:
+                LOG_message(
+                    LOG_source, LOG_SEVERITY_ERROR, LOG_SINK_ALL,
+                    "Failed UART reception."
+                );
+                return 4;   // Error code: Error receiving data
+            case 4:
+                LOG_message(
+                    LOG_source, LOG_SEVERITY_ERROR, LOG_SINK_ALL,
+                    "No response received. Timeout waiting for 1st byte."
+                ); 
+                return 5;   // Error code: Error receiving data
+            case 8:
+                LOG_message(
+                    LOG_source, LOG_SEVERITY_ERROR, LOG_SINK_ALL,
+                    "Not enough space in the MPI response buffer"
+                ); 
+                return 6;   // Error code: Error receiving data
+            default:
+                LOG_message(
+                    LOG_source, LOG_SEVERITY_ERROR, LOG_SINK_ALL,
+                    "Unknown response return: %u", 
+                    MPI_tx_rx_status
+                );
+                return 7;   // Error code: Unhandled error
+        }
+    }
+
+    // UART 4 Selected (CAMERA)
+    else if(strcasecmp(arg_uart_port_name, "CAMERA") == 0) {
```

You need to return and error and say that this is not implemented. Currently it just fails silently.

Some electrical team members wasted a lot of time trying to troubleshoot this UART port tonight, despite not being implemented.

Why isn't this implemented?

---

For the camera since it utilizes DMA very much like the MPI, I had planned to implement it the same way as the MPI with a modularized send/receive function once the Camera code was added to main. My main reasoning for this was to avoid multiple implementations of the same send/receive functionality.

My bad on missing the error message earlier—it’s definitely crucial here.

I can add the basic TX functionality here if that is preferred, I was just a bit optimistic seeing the progress on the Camera code that I decided to just wait on its push before adding it.